homeassistant/components/yamaha/media_player.py

```diff
@@ -191,6 +192,7 @@ async def async_setup_platform(
             _LOGGER.debug("Ignore receiver zone: %s %s", config_info.name, zctrl.zone)
             continue
 
+        assert config_info.name
```

Comment:
Better to change `YamahaConfigInfo` to name not being optional (as it's not optional) given I assume you put this here for type-checking.

Reply:
Well that is the problem, it can happen to be `None` at a given moment in time because of the `.get()`. If I change that to `[]`, I get a KeyError in the tests since it uses the discovery info instead (since `yamaha:` is valid YAML)

Reply:
I actually didn't check but from Frencks comment from before I assumed the schema always guarded that there would be a name. If that's not the case then obviously it should be left like this (but then it will blow instead, potentially already by the debug logging above as it's also using the name)

Reply:
I also tried overwriting the name to be `name: str`, but then I have to add `cast()` to the `.get()`'s and since one of them can be None, I also have to take that into account

Reply:
```python
class YamahaConfigInfo:
    """Configuration Info for Yamaha Receivers."""

    def __init__(
        self, config: ConfigType, discovery_info: DiscoveryInfoType | None
    ) -> None:
        """Initialize the Configuration Info for Yamaha Receiver."""
        self.from_discovery = False
        if discovery_info is not None:
            self.from_discovery = True
            config = discovery_info
        self.name = config[CONF_NAME]
        ...
```
But I don't have the full picture here so maybe I'm proposing something which isn't a workable solution.

Reply:
name is set in the discovery_info block as well